### ISSUE TITLE
string interpolation issues with REPEAT / MACRO

### ISSUE THREAD

**Issue by nonchip** [vhelin/wla-dx#336]
i'm trying to generate reset vectors according to the following pseudo-code:
```
for \R from $00 to $38 step $08 do
  .ORGA \R
  .SECTION "BOOT_RESET_\R" FORCE KEEP
     jp boot_rst_\R
  .ENDS
done
```

what I've tried:
```asm
.REPEAT 8 INDEX IDX
  .ORGA (IDX * $0008)
  .SECTION "BOOT_RESET_IDX" SIZE $0008 FORCE RETURNORG KEEP ; <-- errors
     jp boot_rst_IDX
  .ENDS
.ENDR
```
errors out with:
> DIRECTIVE_ERROR: SECTION "BOOT_RESET_IDX" was defined for the second time.

since there's no string interpolation happening there

so i thought i'd be smart and use a macro instead because those have interpolation:
```asm
.MACRO make_rst
  .ORGA (\1 * $0008)
  .SECTION "BOOT_RESET_\1" SIZE $0008 FORCE RETURNORG KEEP ; <-- errors
     jp boot_rst_\1
  .ENDS
.ENDM
.REPEAT 8 INDEX IDX
make_rst IDX
.ENDR
```
guess what, still errors out with the SAME error, somehow expanding the macro 8 times but replacing `\1` with the string "IDX" instead of the index (i assume the macro expansion happens before the "IDX" gets replaced by its actual value, thus landing me in the same situation as above where the actual string in question is "BOOT_RESET_IDX").

i can live with doing 8 reset vectors by hand, but i'mma be dealing with a rather large IM2 interrupt vector table a bit later, and even without that interpolating a loop index into a string is something one needs all over the place.

tl;dr: can i make `REPEAT ... INDEX` behave like a macro variable? or, even better, string interpolate any definitions?

**Comment by vhelin** [comment]
The "BOOT_RESET_\1" should have expanded to "BOOT_RESET_0", "BOOT_RESET_1", etc., but the string processor just stupidly copied the plain text of the argument to \1... I have no idea why we didn't implement it properly as the string processor has all the information it needs to do it nicely.

Anyway, just fixed it. :) If you now try the latest sources your .MACRO should work as you wanted...

PS. Sorry for the huge delay, I totally missed this issue and only this weekend had lots of free time to go through the pending issues list...

**Comment by nonchip** [comment]
afraid i switched my entire codebase over to `sdcc` in the meantime (who then managed to mess up `IN/OUT` instructions, FUN!) so i can't really test it, also i doubt i'll be wanting to switch back to pure assembler for this project anyway.

**Comment by vhelin** [comment]
No problem, I'm pretty confident I got this one fixed. :)
